### ISSUE TITLE
Freedom issue

### ISSUE THREAD

**Issue by throwaway1037** [yairm210/Unciv#5770]
As far as I can tell, Unciv is not 100% [libre](https://freedomdefined.org).

To package in a 100% libre OS like [Parabola](https://parabola.nu), the game must be 100% libre.

As such, please may all the proprietary data be replaced with libre replacements, or at least clearly separated from the libre data, so that it can be omitted and replaced with placeholders for the 100% libre build of the game?

Also, does Unciv include any proprietary code, proprietary dependencies, or otherwise integrate with proprietary software? The README mentions Dropbox; can this functionality be easily disabled with a build option at compile-time?

Thank you very much.

**Comment by yairm210** [comment]
A couple of points here
A. Adding flags to disable functionality is possible
B. Anything that is included in LibGDX is *not* replaceable - is that considered "libre"?

**Comment by throwaway1037** [comment]
A. Please may I request this as a feature?
B. Yes, LibGDX is libre because it is released under the Apache 2.0 license, which is a [free software license](https://gnu.org/licenses/license-list.html).

**Comment by yairm210** [comment]
> A. Please may I request this as a feature? B. Yes, LibGDX is libre because it is released under the Apache 2.0 license, which is a [free software license](https://gnu.org/licenses/license-list.html).

Unciv is also released under MPL 2.0, which is in that same list, so I'm not sure how you classify them as different


> Checking the credits file, it seems that somebody needs to make a table and document the license of each file, so that the proprietary files can be replaced or removed. Since there are dozens of files, this could be a lot of work, and it's likely that there will be discrepancies between the list and the actual files in the repository since it's inevitable that someone made a mistake somewhere when dealing with dozens of files.

Does Creative Commons not satisfy "libre"? And if it does, then what difference does it make? All items there are either CC or public domain

**Comment by throwaway1037** [comment]
> so I'm not sure how you classify them as different

Libraries like LibGDX are usually mostly code, so the project license usually covers the entire project, whereas games usually include tons of other data, like graphics, audio, data files, etc., and different types of files are usually licensed differently, eg. It's not recommended to use a license published by Cteative Commons for code, and fonts are usually licensed under the SIL OFL.

This means that games usually have lots of different files under lots of different licenses, so there's a much larger possibility for files to be licensed under a proprietary license.

> Does Creative Commons not satisfy libre?

Creative Commons is not a license, but an organisation that publishes licenses. Some of their licenses are libre, and some are not; they have published dozens of licenses in the past.

> All items are either CC...

As such, it doesn't really make sense to say this. Instead it would be better to specify exactly which license each file is licensed under, so that every user knows their rights, eg. File XYZ is licensed under CC-BY-SA-4.0-Int-or-later.

**Comment by xlenstra** [comment]
We use dropbox as a 'server' for multiplayer games and we have no easy replacement of that (see also: #2679). Removing that is possible, but would thus basically remove the entirety of the multiplayer feature.

As for the assets mentioned in the credits, most (but not all) of the images are from `thenounproject.com`, all of which have either a CC-BY or a PD license (according to their website). Almost all sounds are from `freesound.org`, which looks like it ditributes all images either under CC, CC-BY or CC-BY-NC (according to their website). Apart from these there are probably like 20-40 assets from other sources, checking them manually is doable but a lot of work.

**Comment by throwaway1037** [comment]
Removing Dropbox integration is necessary in order to liberate the game because:
- the official Dropbox client is proprietary (although I think Unciv doesn't need it?)
- the TOS/Ts&Cs/Privacy Policy are odious
- feding data through Dropbox is a privacy issue

May I recommend switching to a decentralised, P2P model of networking, so dedicated centralised servers are not necessary? This would require effort to rewrite code, but would pay off in the long term, as decentralised networks have many advantages over centralised ones, such as:
- censorship resistance
- improved privacy and security (no data flowing through a central server) 
- more robust (no single point of failiure)

For the time being, it would be acceptable to sacrifice the multiplayer feature via a build option, if that's what it takes to remove Dropbox integration.

> assets

I'd rather not refer to published works as such, because it [treats them merely as cash cows instead of something to be appreciated on their own merits](https://gnu.org/philosophy/words-to-avoid.html#Assets).

Nevertheless, it would indeed take a lot of work to manually document the licenses of each file and either replace the proprietary ones, or omit them from a 100% libre build via a build option. Also, the NC-type licenses are [not libre](https://freedomdefined.org/FAQ#Why_isn.27t_a_Non-Commercial_restriction_considered_free.3F); see [this article](https://freedomdefined.org/Licenses/NC) for the case against them.

**Comment by yairm210** [comment]
In short, this sounds like a lot of boring, repetitive work for a tiny subset of users.
You're welcome to do this yourself and submit a PR, but this is not the story of task I wish to dedicate my free time to, and I do not see it as part of the product roadmap

**Comment by will-ca** [comment]
Currently illegal files/sources (ignoring fair use):

https://www.soundjay.com/fire-sound-effects.html (Reuploading is forbidden, which this GH repository is.)

http://eab.abime.net/showthread.php?t=67078&styleid=4 (I think this is just a rip of a commercial game?)

https://filmmusic.io (Improper attribution— Are the affected files even still in the game?)

https://www.pngimages.pics/275495/transparent-medieval-battering-ram-png.php (Despite the "Download" button, site does not provide license.)

https://www.vectorstock.com/royalty-free-vector/flat-icon-design-collection-anti-aircraft-gun-vector-16396060 (Derivatives and distribution both forbidden.)

https://www.pngkit.com/view/u2w7i1i1o0t4a9y3_png-file-merchant-navy-icon/ (Seems the site wants to release it for free, but neglected to add an explicit grant of permissions, so by default no uses are allowed.)

https://www.shutterstock.com/image-vector/maori-symbol-spiral-shape-based-on-1145645057?id=1145645057 (Redistribution forbidden.)

https://github.com/ravignir/5Hex-Tileset/blob/master/Images/ (@ravignir License file? Otherwise all use is technically illegal. Don't know if same applies to files by JackRainy, or if they explicitly specified a license. If they explicitly submitted the files, then I'd call that good enough.)

List may be non-exhaustive. Additional files may be "non-free" but currently legal.

Recommend using a REGEX to match all URLs that aren't TheNounProject.Com or Freesound: `http[s]{0,1}://(?!(thenoun|freesound))`.

Aside from the above, technically outright illegal files, the other "non-free" sources that jumped out to me were:
- FlatIcon.Com (although `Credits.md` says the specific author for most of the files released under CC-BY-3.0, which I could not verify).
- Potential CC-NC content from Freesound (which looks very easy to write a Python scraper for).

Additional files unaccounted for in `Credits.md` may exist.

I am not a lawyer.

**Comment by yairm210** [comment]
Nice! That is definitely actionable, so I'm reopening this issue

**Comment by yairm210** [comment]
Ravignir made his tileset himself, so that's definitely not a problem

**Comment by yairm210** [comment]
It seems the comment was deleted 0_0
Oh well
Can always be either reopened or opened as a separate issue

**Comment by xlenstra** [comment]
The single advantage of having email notifications of everything that happens on this GitHub is that even deleted messages can still be read :)
Though I respect the previous' author's decision of removing this (whatever their reason might be), I'll go over the 'illegal' links mentioned to verify the legality of us using them sometime this week.

**Comment by will-ca** [comment]
Oops. I saw the issue was closed basically right as I finished writing the comment, and it'll still be a fair bit of work to double-check and replace the sources I mentioned even with a list, so I figured I'd rather sort it into my "May PR" pile than raise any noise about it. Anyway, I still kept a copy of the comment in my notes for that:

> Currently illegal files/sources (ignoring fair use):
> 
> https://www.soundjay.com/fire-sound-effects.html (Reuploading is forbidden, which this GH repository is.)
> 
> http://eab.abime.net/showthread.php?t=67078&styleid=4 (I think this is just a rip of a commercial game?)
> 
> https://filmmusic.io (Improper attribution— Are the affected files even still in the game?)
> 
> https://www.pngimages.pics/275495/transparent-medieval-battering-ram-png.php (Despite the "Download" button, site explicitly does not provide license.)
> 
> https://www.vectorstock.com/royalty-free-vector/flat-icon-design-collection-anti-aircraft-gun-vector-16396060 (Derivatives and distribution both forbidden.)
> 
> https://www.pngkit.com/view/u2w7i1i1o0t4a9y3_png-file-merchant-navy-icon/ (Seems the site wants to release it for free, but neglected to add an explicit grant of permissions, so by default no uses are allowed.)
> 
> https://www.shutterstock.com/image-vector/maori-symbol-spiral-shape-based-on-1145645057?id=1145645057 (Redistribution forbidden.)
> 
> https://github.com/ravignir/5Hex-Tileset/blob/master/Images/ (@ravignir License file? Otherwise all use is technically illegal. Don't know if same applies to files by JackRainy, or if they explicitly specified a license. If they explicitly submitted the files, then I'd probably call that good enough.)
> 
> List may be non-exhaustive. Additional files may be "non-free" but currently legal.
> 
> Recommend using a REGEX to match all URLs that aren't TheNounProject.Com or Freesound: `http[s]{0,1}://(?!(thenoun|freesound))`.
> 
> Aside from the above, technically outright illegal files, the other "non-free" sources that jumped out to me were:
> - FlatIcon.Com (although `Credits.md` says the specific author for most of the files released under CC-BY-3.0, which I could not verify).
> - Potential CC-NC content from Freesound (which looks very easy to write a Python scraper for).
> 
> Additional files unaccounted for in `Credits.md` may exist.
> 
> I am not a lawyer.

Regarding the 5Hex assets, I mean that unless Ravignir explicitly gave permission (such as by releasing under CC), then we're *technically*/legally still not able to use or relicense his art. Not problem, but more like covering all bases.

**Comment by xlenstra** [comment]
> Oops. I saw the issue was closed basically right as I finished writing the comment, and it'll still be a fair bit of work to double-check and replace the sources I mentioned even with a list, so I figured I'd rather sort it into my "May PR" pile than raise any noise about it. Anyway, I still kept a copy of the comment in my notes for that:
> 
> > Currently illegal files/sources (ignoring fair use):
> > https://www.soundjay.com/fire-sound-effects.html (Reuploading is forbidden, which this GH repository is.)

If I read their terms of service correctly (but I am no lawyer so I could be wrong), it is illegal to post them on a place where the main purpose is for users to download them or use them for their own projects. As we only use the files to create the game, I'd say that is a different use case and thus falls under the allowance to use the sound excerpts in games.

> > http://eab.abime.net/showthread.php?t=67078&styleid=4 (I think this is just a rip of a commercial game?)

I'm unable to find any trees in that post that are still part of the current jungle tiles. So either they were changed over the course of time, or only inspiration was taken from these with Ravignir creating their own version? I'd say this isn't a real problem.

> > https://filmmusic.io (Improper attribution— Are the affected files even still in the game?)

This was originally used for the only music track before SomeTroglodyte made their great way to add moddable music. I think that if you have not downloaded any of the mods, it is still the default soundtrack that plays. I've changed the attribution to point to the correct file

> > https://www.pngimages.pics/275495/transparent-medieval-battering-ram-png.php (Despite the "Download" button, site explicitly does not provide license.)

Yup, according to their terms, https://www.pngimages.pics/terms.php , we are using this image illeagly. I'll find a replacement.

> > https://www.vectorstock.com/royalty-free-vector/flat-icon-design-collection-anti-aircraft-gun-vector-16396060 (Derivatives and distribution both forbidden.)

Illegal as well, I'll look for a replacement. 

> > https://www.pngkit.com/view/u2w7i1i1o0t4a9y3_png-file-merchant-navy-icon/ (Seems the site wants to release it for free, but neglected to add an explicit grant of permissions, so by default no uses are allowed.)

Looking further, this same image is released in other places online, but none of these allow it to be downloaded. Should be replaced as well.

> > https://www.shutterstock.com/image-vector/maori-symbol-spiral-shape-based-on-1145645057?id=1145645057 (Redistribution forbidden.)

Correct, and this also seems to behind a paywall I'm not sure who paid for. Should be replaced.

> > https://github.com/ravignir/5Hex-Tileset/blob/master/Images/ (@ravignir License file? Otherwise all use is technically illegal. Don't know if same applies to files by JackRainy, or if they explicitly specified a license. If they explicitly submitted the files, then I'd probably call that good enough.)
> > List may be non-exhaustive. Additional files may be "non-free" but currently legal.
> > Recommend using a REGEX to match all URLs that aren't TheNounProject.Com or Freesound: `http[s]{0,1}://(?!(thenoun|freesound))`.
> > Aside from the above, technically outright illegal files, the other "non-free" sources that jumped out to me were:
> > 
> > * FlatIcon.Com (although `Credits.md` says the specific author for most of the files released under CC-BY-3.0, which I could not verify).
> > * Potential CC-NC content from Freesound (which looks very easy to write a Python scraper for).
> > 
> > Additional files unaccounted for in `Credits.md` may exist.
> > I am not a lawyer.
> 
> Regarding the 5Hex assets, I mean that unless Ravignir explicitly gave permission (such as by releasing under CC), then we're _technically_/legally still not able to use or relicense his art. Not a problem, but more like covering all bases.

**Comment by SomeTroglodyte** [comment]
I remember arguing against soundjay when it was added. A permissive sounding site is worth nothing, a license check must be doable straightforward and the result must be explicit and unambiguous.

And as far as I remember the filmmusic.io / incompetech _sites_ have changed in the last 18 months or so, which makes some attributions seem incorrect - they were / could have been at the time of writing. The [latest credits I did for them](https://github.com/SomeTroglodyte/Additional-Music-Ambient/blob/master/credits.md) in mods are somewhat different - better than here???

:+1: :+1: I'm all for slowly making our way to a fully libre project. Let's at least pay attention to stay libre with any new content! :+1: :+1:

**Comment by throwaway1037** [comment]
@xlenstra thank you very much for #5819.

@SomeTroglodyte thank you for your support; I agree Unciv should become 100% libre.

**Comment by refola** [comment]
This issue reminded me that I didn't explicitly license and wasn't directly attributed for the `unciv.desktop` and `Unciv.sh` files I posted on the (now defunct) CCR as linked to in my opening of #1380 and committed by @yairm210 in 8854caf.

For simplicity, I'm explicitly licensing said contributions (which are minor enough that I would want "fair use" to apply if copyright law were sane) contributions (and the `PKGBUILD` which seems to have a derivative in Arch's AUR now) as your choice of public domain, [https://creativecommons.org/share-your-work/public-domain/cc0/ CC0], and MPL 2.0+ (per [https://github.com/yairm210/Unciv/blob/master/LICENSE Unciv's overall `LICENSE`]). Also, for the sake of completion, I disclaim all warranty, express or implied.

Tl;dr: Two files that were missed are now properly licensed.
